pandas/tests/plotting/frame/test_frame.py

```diff
@@ -787,6 +787,25 @@ def test_plot_scatter_with_s(self):
         ax = df.plot.scatter(x="a", y="b", s="c")
         tm.assert_numpy_array_equal(df["c"].values, right=ax.collections[0].get_sizes())
 
+    def test_plot_scatter_with_norm(self):
+        # added while fixing GH 45809
+        import matplotlib as mpl
+
+        df = DataFrame(np.random.random((10, 3)) * 100, columns=["a", "b", "c"])
+        norm = mpl.colors.LogNorm()
+        ax = df.plot.scatter(x="a", y="b", c="c", norm=norm)
+        assert ax.collections[0].norm is norm
+
+    def test_plot_scatter_without_norm(self):
+        # added while fixing GH 45809
+        import matplotlib as mpl
+
+        df = DataFrame(np.random.random((10, 3)) * 100, columns=["a", "b", "c"])
+        ax = df.plot.scatter(x="a", y="b", c="c")
+        color_min_max = (df.c.min(), df.c.max())
+        default_norm = mpl.colors.Normalize(*color_min_max)
+        assert all(df.c.apply(lambda x: ax.collections[0].norm(x) == default_norm(x)))
```

Comment:
Could this be done without apply? It will be easier to debug if this assertion doesn't go through necessary code paths.

Reply:
What's the better approach? Just a loop?

Reply:
Yeah the old check you had prior was good. The data doesn't have to be random as well if that helps shorten your loop and allows looping over less values

Reply:
Ok, I'll revert to the previous check. I think looping over the values in the dataframe makes sense; it's replicating what's going on in the actual plot, and there's no real benefit to including values that are out of bounds - that's matplotlib normalize functionality.

Reply:
Changes made